contracts/BoostPassVotes.sol

```diff
@@ -0,0 +1,209 @@
+    }
+
+    function getPastVotes(address account, uint256 timepoint) public view returns (uint256) {
+        uint48 currentTimepoint = clock();
+        if (timepoint >= currentTimepoint) {
+            revert ERC5805FutureLookup(timepoint, currentTimepoint);
+        }
+
+        uint256 pastVotes = _delegateCheckpoints[account].upperLookupRecent(SafeCastLib.toUint32(timepoint));
+
+        if (pastVotes != 0) {
+            return pastVotes;
+        }
+
+        return this.getVotes(account);
+    }
+
+    function getPastTotalSupply(uint256 timepoint) public view virtual override returns (uint256) {
+        uint48 currentTimepoint = clock();
+        if (timepoint >= currentTimepoint) {
+            revert ERC5805FutureLookup(timepoint, currentTimepoint);
+        }
+
+        if (timepoint <= boostPassCheckpoint) {
+            return boostPassSupplyAtCheckpoint;
+        }
+
+        uint256 pastTotalSupply = _totalCheckpoints.upperLookupRecent(SafeCastLib.toUint32(timepoint));
+        
+        if (pastTotalSupply != 0) {
+            return pastTotalSupply;
+        }
+
+        return BoostPass(boostPassAddress).totalSupply();
+    }
+
+    function delegates(address account) public view returns (address) {
+        return _delegation[account];
+    }
+
+    function delegate(address delegatee) public {
+        address account = _msgSender();
+        _delegate(account, delegatee);
+    }
+
+    function delegateBySig(address delegatee, uint256 nonce, uint256 expiry, uint8 v, bytes32 r, bytes32 s) public {
+        if (block.timestamp > expiry) {
+            revert VotesExpiredSignature(expiry);
+        }
+        address signer = Ecdsa.recover(
+            _hashTypedDataV4(keccak256(abi.encode(_DELEGATION_TYPEHASH, delegatee, nonce, expiry))),
+            v,
+            r,
+            s
+        );
+        require(nonce == _useNonce(signer), "Votes: invalid nonce");
+        _delegate(signer, delegatee);
+    }
+
+    function _delegate(address account, address delegatee) internal virtual {
+        address oldDelegate = delegates(account);
+        _delegation[account] = delegatee;
+
+        emit DelegateChanged(account, oldDelegate, delegatee);
+        _moveDelegateVotes(oldDelegate, delegatee, _getVotingUnits(account));
+    }
+
+    function _transferVotingUnits(address from, address to, uint256 amount) internal virtual {
+        if (from == address(0)) {
+            _push(_totalCheckpoints, _add, SafeCastLib.toUint224(amount));
+        }
+        if (to == address(0)) {
+            _push(_totalCheckpoints, _subtract, SafeCastLib.toUint224(amount));
+        }
+        // we don't need to move delegate votes upon mint
+        // _moveDelegateVotes(delegates(from), delegates(to), amount);
+    }
+
+    function _moveDelegateVotes(address from, address to, uint256 amount) private {
+        if (from != to && amount > 0) {
+            if (from != address(0)) {
```

Comment:
If we use a separate internal function for the case where address(0) is to/from we could avoid needing to do these checks. Something about this logic flow feels off to me. If we keep it as it is I'd add a comment/callout for when and why address(0) is. Something like "When initially minting we don't want to change the delegated votes of address(0)" idk.